tests/system/Email/EmailTest.php

```diff
@@ -139,4 +136,30 @@ public function testFailureDoesNotTriggerEvent()
 
         $this->assertNull($result);
     }
+
+    public function testDestructDoesNotThrowException()
+    {
+        $email = $this->getMockBuilder(Email::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['sendCommand'])
+            ->getMock();
+        $email->method('sendCommand')
+            ->willThrowException(new ErrorException('SMTP Error.'));
+
+        // Force resource to be injected into the property
+        $SMTPConnect = fopen(__FILE__, 'rb');
+        $this->setPrivateProperty($email, 'SMTPConnect', $SMTPConnect);
+
+        $email->__destruct();
+
+        $this->assertTrue(true);
```

Comment:
Is this necessary? If I remember, the invocation call using `will*()` (combined with `expects()`)  is effectively an assertion.

Reply:
`expects($this->any())` causes still the warning, but `$this->once()` works fine.